ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -419,16 +442,40 @@ impl UnverifiedValuesStore {
         }
     }
 
-    /// `get_unverified_field_value` converts a `UnverifiedValuesStore` into an iterator over `UnverifiedFieldValues`
-    /// compatible with DZKPs
-    #[allow(dead_code)]
-    fn get_unverified_field_values<DF: DZKPBaseField>(
+    /// `get_field_values_prover` converts a `UnverifiedValuesStore` into an iterator over `field`
+    /// values used by the prover of the DZKPs
+    #[cfg(all(test, unit_test))]
+    fn get_field_values_prover<DF: DZKPBaseField>(
+        &self,
+    ) -> impl Iterator<Item = UVPolynomials<DF>> + '_ {
+        self.vec
+            .iter()
+            .flatten()
+            .flat_map(UnverifiedValues::convert_prover::<DF>)
+    }
```

Comment:
I think this API is overly complicated. There are nearly identical looking methods on `UnifiedValuesStore` and `Batch`, and I don't know why we need both. Can we eliminate a lot of code by reducing the number of types?

Reply:
I don't see how to eliminate them, this is also not related to this PR. Batch is a wrapper around a HashMap and plays a similar role as AccumulatorState in the MAC based approach. We use Arc and weak pointer to Batch (and AccumulatorState). The HashMap uses Gates as Keys and under each Gate sits a UnverifierdValuesStore that collects the multiplication bits for that gate. It is important to separate them by gate since we need to have a consistent ordering of the multiplications. Further, different gates might use differently sized bitarray types. So it makes sense to group them by gate.